postfix/check.py

```diff
@@ -49,6 +58,50 @@ def _get_config(self, instance):
 
         return instance_config
 
+    def _get_postqueue_stats(self, tags):
+        # postqueue gathers information for active, hold and deferred queue.
+        #
+        #     Each  queue entry shows the queue file ID, message size, arrival
+        #     time, sender, and the recipients that still need  to  be  deliv-
+        #     ered.  If mail could not be delivered upon the last attempt, the
+        #     reason for failure is shown. The queue ID string is followed  by
+        #     an optional status character:
+        #
+        #     *      The  message  is in the active queue, i.e. the message is
+        #            selected for delivery.
+        #
+        #     !      The message is in the hold queue, i.e. no further  deliv-
+        #            ery  attempt  will  be  made  until the mail is taken off
+        #            hold.
+        #
+
+        postfix_conf, _, _ = get_subprocess_output(['postconf', 'mail_version'], self.log, False)
+        postfix_version = postfix_conf.strip('\n').split('=')[1].strip()
+        output, _, _ = get_subprocess_output(['postqueue', '-p'], self.log, False)
+
+        active_count = 0
+        hold_count = 0
+        deferred_count = 0
+
+        for line in output.splitlines():
+            if '*' in line:
+                active_count += 1
+                continue
+            if '!' in line:
+                hold_count += 1
+                continue
+            if line[0:1].isdigit():
+                deferred_count += 1
+        self.log.debug('Postfix Version: %s' % postfix_version)
+        self.log.debug('active_count: %d' % active_count)
```

Comment:
Not sure we need to log `active_count`, `hold_count` and `deferred_count` to be honest - we can see those values on the UI. Postfix version is 👍